### ISSUE TITLE
Metatdata truncate the header values with comma

### ISSUE THREAD

**Issue by shanshanzhu** [grpc/grpc-node#769]
### Problem description

When we log headers in call-stream.ts here:
https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/call-stream.ts#L213

We see the field
```
{content-type":"application/grpc","date":"Thu, 07 Mar 2019 21:24:38 GMT"}
```

However, when we log the metadata in our code:
```
call.on('metadata', metadata => { console.log(metadata); })
```

We see the log
```
{ 'content-type': 'application/grpc', date: 'Thu' }
```

I think this is caused by this line of code https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/metadata.ts#L219

### Reproduction steps
Give very precise steps you've discovered to reproduce your problem. If possible and applicable, provide us with a repository we can clone that contains a reproduction case. Also if possible and applicable, please include a Dockerfile that exhibits the problem if it's specific to a certain environment. Bug reports with no reproduction steps will be closed.

### Environment
 - OS name, version and architecture: [macOS Mojave 10.14]
 - Node version [10.15.0]
 - Node installation method [ nvm]
 - Package name and version [@grpc/grpc-js @0.3.2]

### Additional context
It should be straightforward to add this test cases to `fromHttp2Headers` in `metadata_test`

**Comment by murgatroid99** [comment]
This is intended behavior. The following quotation from the [protocol specification document](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md) is relevant:

> **Custom-Metadata** header order is not guaranteed to be preserved except for values with duplicate header names. Duplicate header names may have their values joined with "," as the delimiter and be considered semantically equivalent.

So, that header is being treated as two separate "date" headers, one with the value "Thu" and the other with the value "07 Mar 2019 21:24:38 GMT". I don't know what exactly you're getting when you do `console.log(metadata)`, but both values should be in the actual object

I will admit that this behavior is not mentioned in the Node library's documentation, and it probably should be.

**Comment by murgatroid99** [comment]
A correction to my previous comment: I misinterpreted the specification that I quoted above. Splitting headers this way is not correct behavior and this is a bug in grpc-js. A fix should be out for this soon.

**Comment by murgatroid99** [comment]
I published the fix for this in grpc-js version 1.0.4.
